packages/babel-standalone/src/index.js

```diff
@@ -136,97 +136,37 @@ export function registerPresets(newPresets) {
-  "transform-es2015-object-super": require("babel-plugin-transform-es2015-object-super"),
-  "transform-es2015-parameters": require("babel-plugin-transform-es2015-parameters"),
-  "transform-es2015-shorthand-properties": require("babel-plugin-transform-es2015-shorthand-properties"),
-  "transform-es2015-spread": require("babel-plugin-transform-es2015-spread"),
-  "transform-es2015-sticky-regex": require("babel-plugin-transform-es2015-sticky-regex"),
-  "transform-es2015-template-literals": require("babel-plugin-transform-es2015-template-literals"),
-  "transform-es2015-typeof-symbol": require("babel-plugin-transform-es2015-typeof-symbol"),
-  "transform-es2015-unicode-regex": require("babel-plugin-transform-es2015-unicode-regex"),
-  "transform-es3-member-expression-literals": require("babel-plugin-transform-es3-member-expression-literals"),
-  "transform-es3-property-literals": require("babel-plugin-transform-es3-property-literals"),
-  "transform-es5-property-mutators": require("babel-plugin-transform-es5-property-mutators"),
-  "transform-eval": require("babel-plugin-transform-eval"),
-  "transform-exponentiation-operator": require("babel-plugin-transform-exponentiation-operator"),
-  "transform-export-extensions": require("babel-plugin-transform-export-extensions"),
-  "transform-flow-comments": require("babel-plugin-transform-flow-comments"),
-  "transform-flow-strip-types": require("babel-plugin-transform-flow-strip-types"),
-  "transform-function-bind": require("babel-plugin-transform-function-bind"),
-  "transform-jscript": require("babel-plugin-transform-jscript"),
-  "transform-object-assign": require("babel-plugin-transform-object-assign"),
-  "transform-object-rest-spread": require("babel-plugin-transform-object-rest-spread"),
-  "transform-object-set-prototype-of-to-assign": require("babel-plugin-transform-object-set-prototype-of-to-assign"),
-  "transform-proto-to-assign": require("babel-plugin-transform-proto-to-assign"),
-  "transform-react-constant-elements": require("babel-plugin-transform-react-constant-elements"),
-  "transform-react-display-name": require("babel-plugin-transform-react-display-name"),
-  "transform-react-inline-elements": require("babel-plugin-transform-react-inline-elements"),
-  "transform-react-jsx": require("babel-plugin-transform-react-jsx"),
-  "transform-react-jsx-compat": require("babel-plugin-transform-react-jsx-compat"),
-  "transform-react-jsx-self": require("babel-plugin-transform-react-jsx-self"),
-  "transform-react-jsx-source": require("babel-plugin-transform-react-jsx-source"),
-  "transform-regenerator": require("babel-plugin-transform-regenerator"),
-  "transform-runtime": require("babel-plugin-transform-runtime"),
-  "transform-strict-mode": require("babel-plugin-transform-strict-mode"),
+// Bundle all the plugins included in the Babel repo
+const pluginContext = require.context(
+  __dirname + "/../../",
```

Comment:
I see that these are all in https://github.com/babel/babel/blob/7.0/packages/babel-standalone/package.json#L17. I think I'd have expected this to package the ones available in `node_modules` rather than look at `packages/*` so that we have a clear whitelist of what is included.

Thoughts? I definitely admit it's certainly easier to just grab everything, but on the other hand there's a certain nice clarity to having them be normal package dependencies.

Reply:
>  I think I'd have expected this to package the ones available in node_modules rather than look at packages/* so that we have a clear whitelist of what is included.

I think the node_modules ones would be hoisted to the root (as all packages in the repo would depend on the versions in the repo), so simply doing `./node_modules` wouldn't work. I could do `../../../../node_modules/` or whatever, but I'm not sure if there's any real advantage to doing that given all the dependencies are in there.

Reply:
Hmm you may be right. I was thinking `node_modules` would be all symlinks, but I guess that's not what we have anymore.

One other case I worry about if we're not explicitly listing deps is what happens if we decide to stop publishing new versions of something and remove it from the monorepo. We'll have to explicitly add it to this list to not break things, but that's going to be pretty hard to remember to do.

Reply:
> I was thinking node_modules would be all symlinks, but I guess that's not what we have anymore.

I think any good hoisting algorithm would hoist the packages to the top-level `node_modules` directory in this case (everything references a single version of the packages).

> if we decide to stop publishing new versions of something and remove it from the monorepo.

Isn't that semver-major since it's a breaking change? Also if people need the old plugin, they could just stay with the old version? Not saying that's a good solution, but are there any cases where we'd remove a plugin without a major (or at least a minor) version bump?

We could solve that by bundling stuff from both `packages` and `node_modules` if you think it's worth doing that.

Reply:
> Isn't that semver-major since it's a breaking change?

If we just decide not to release new versions of a package, it's not breaking anything, we're just saying we don't care about making future releases of it. I can't say off the top of my head.

My concern mostly is that by having a whitelist, we don't have to worry that changes to what's in the monorepo will accidentally affect standalone. Changes to it have to be on purpose. It seems like your goal is the opposite in favor of less grunt work when adding new stuff, I assume?

To me the explicit list seems to add a lot of clarity, and keeps the package from having to know anything about its outer structure, which is something the other packages all stick to. I think I'm a little hesitant to break the boundary by having things in `packages` look at parent folder structure.

Reply:
That all makes sense. Let's keep the whitelist then? 😃 

I might just add a comment linking to this PR, for posterity. Something like:

```
// Want to get rid of this long whitelist of plugins/presets?
// Wait! Please read https://github.com/babel/babel/pull/6177 first.
```

Reply:
Go for it :)

Reply:
https://github.com/babel/babel/pull/6910